app/characters/knight.py

```diff
@@ -21,23 +21,18 @@ def __init__(
         self.protection = protection
 
     def get_armour(self) -> None:
-        self.protection = 0
-        for elem in self.armour:
-            self.protection += elem["protection"]
+        self.protection = sum([elem["protection"] for elem in self.armour])
 
     def get_weapon(self) -> None:
         self.power += self.weapon["power"]
 
-    def get_potion(self) -> None:
-        if self.potion is not None:
-            if "power" in self.potion["effect"]:
-                self.power += self.potion["effect"]["power"]
-
-            if "protection" in self.potion["effect"]:
-                self.protection += self.potion["effect"]["protection"]
-
-            if "hp" in self.potion["effect"]:
-                self.hp += self.potion["effect"]["hp"]
+    def get_potion(self) -> int | None:
+        if self.potion is None:
+            return 0
+        effect = self.potion.get("effect", {})
+        self.power += effect.get("power", 0)
+        self.protection += effect.get("protection", 0)
+        self.hp += effect.get("hp", 0)
```

Comment:
looks much more better ;)
You can read about Cognitive complexity more(it's the issue you had here)